src/backend/vulkan/src/window.rs

```diff
@@ -368,6 +449,14 @@ impl hal::Surface<Backend> for Surface {
     }
 }
 
+impl Surface {
+    pub(crate) fn update_extent(&mut self, physical_device: &PhysicalDevice) {
+        use hal::Surface;
+        let (caps, _) = self.capabilities_and_formats(physical_device);
```

Comment:
is there a reason we can't generate `caps` during initialization and just keep it instead of re-generating here?

Reply:
Because the extent changes when the window resizes? How else do I query for the new size?